qiskit_runtime/vqe/vqe_program.py

```diff
@@ -0,0 +1,362 @@
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Qiskit Nature VQE Quantum Program."""
+
+
+from typing import List, Union, Callable, Optional, Any, Dict
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.exceptions import QiskitError
+from qiskit.providers import Provider
+from qiskit.providers.backend import Backend
+from qiskit.algorithms import MinimumEigensolver, MinimumEigensolverResult, VQEResult
+from qiskit.opflow import OperatorBase, PauliSumOp
+from qiskit.quantum_info import SparsePauliOp
+
+
+class VQEProgram(MinimumEigensolver):
+    """The Qiskit Nature VQE Quantum Program to call the VQE runtime as a MinimumEigensolver."""
+
+    def __init__(self,
+                 ansatz: QuantumCircuit,
+                 optimizer: Optional[Dict[str, Any]] = None,
+                 initial_point: Optional[Union[List, np.ndarray]] = None,
+                 provider: Optional[Provider] = None,
+                 backend: Optional[Backend] = None,
+                 shots: int = 1024,
+                 readout_error_mitigation: bool = True,
+                 callback: Optional[Callable[[int, np.ndarray, float, float], None]] = None,
+                 store_intermediate: bool = False,
+                 ) -> None:
+        """
+        Args:
+            ansatz: A parameterized circuit used as Ansatz for the wave function.
+            optimizer: A dictionary specifying a classical optimizer.
+                Currently only SPSA and QN-SPSA are supported. Per default, SPSA is used.
+                The dictionary must contain a key ``name`` for the name of the optimizer and may
+                contain additional keys for the settings.
+                E.g. ``{'name': 'SPSA', 'maxiter': 100}``.
+            backend: The backend to run the circuits on.
+            initial_point: An optional initial point (i.e. initial parameter values)
+                for the optimizer. If ``None`` a random vector is used.
+            shots: The number of shots to be used
+            readout_error_mitigation: Whether or not to use readout error mitigation.
+            callback: a callback that can access the intermediate data during the optimization.
+                Four parameter values are passed to the callback as follows during each evaluation
+                by the optimizer for its current set of parameters as it works towards the minimum.
+                These are: the evaluation count, the optimizer parameters for the
+                ansatz, the evaluated mean and the evaluated standard deviation.
+            store_intermediate: Whether or not to store intermediate values of the optimization
+                steps. Per default False.
+        """
+        if optimizer is None:
+            optimizer = {'name': 'SPSA'}
+
+        # define program name
+        self._program_name = 'vqe'
```

Comment:
nit - can we use `program_id` instead of `program_name`? A program has both a name and an ID and they can be different (they're all the same for the initial release but that'll likely change in the future). Program ID is the one needed for `runtime.run()`.

Reply:
sure!